From bf71e002fb1445f67fd57c8d40409714b8731152 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Thu, 3 Mar 2011 18:04:40 +0800
Subject: [PATCH] ENGR00140122 MX53_SMD: p1003: report single pointer.

Since Ubuntu needs report single pointer, I add BTN_TOUCH
event in single pointer event report.

It will not affact Android, also made ubuntu touchscreen works.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 drivers/input/touchscreen/p1003_v33-ts.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/input/touchscreen/p1003_v33-ts.c b/drivers/input/touchscreen/p1003_v33-ts.c
index ede2e29..0f02732 100644
--- a/drivers/input/touchscreen/p1003_v33-ts.c
+++ b/drivers/input/touchscreen/p1003_v33-ts.c
@@ -147,6 +147,9 @@ static void p1003_work(struct work_struct *work)
 			input_event(input, EV_ABS, ABS_MT_POSITION_Y, y1);
 			input_event(input, EV_ABS, ABS_MT_TOUCH_MAJOR, 1);
 			input_mt_sync(input);
+			input_event(input, EV_ABS, ABS_X, x1);
+			input_event(input, EV_ABS, ABS_Y, y1);
+			input_event(input, EV_KEY, BTN_TOUCH, 1);
 			input_sync(input);
 			old_state->x1 = x1;
 			old_state->y1 = y1;
@@ -186,6 +189,7 @@ static void p1003_work(struct work_struct *work)
 	 * release to user space. */
 	input_event(input, EV_ABS, ABS_MT_TOUCH_MAJOR, 0);
 	input_mt_sync(input);
+	input_event(input, EV_KEY, BTN_TOUCH, 0);
 	input_sync(input);
 	old_state->state = data[0];
 }
@@ -276,6 +280,9 @@ static int __devinit p1003_probe(struct i2c_client *client,
 	__set_bit(EV_ABS, input_dev->evbit);
 	__set_bit(EV_KEY, input_dev->evbit);
 	__set_bit(BTN_TOUCH, input_dev->keybit);
+	__set_bit(ABS_X, input_dev->absbit);
+	__set_bit(ABS_Y, input_dev->absbit);
+	__set_bit(ABS_PRESSURE, input_dev->absbit);
 
 	input_set_abs_params(input_dev, ABS_MT_POSITION_X, 0, xmax, 0, 0);
 	input_set_abs_params(input_dev, ABS_MT_POSITION_Y, 0, ymax, 0, 0);
-- 
1.5.4.4

